denote-to-logseq.el

```diff
@@ -0,0 +1,63 @@
+;;; package -- Utilities to migrate denote notes to logseq
+;;
+;; Author: hapst3r
+;; Version: 0.1
+;; URL: https://github.com/bitspook/notes-migrator/denote-to-logseq.el
+;; Keywords: Denote, Logseq
+;; Version: 0.1.0
+;; Package-Requires: ((emacs "27.2") (denote "1.0.0")
+;;
+;;; Commentary:
+;; A WIP package for migrating Denote notes to Logseq
+;;
+;;; Code:
+
+(require 'denote)
+
+(defvar logseq-directory nil)
+
+(defun nm--convert-denote-links-to-logseq ()
+  "Convert all denote links to logseq links in current buffer.
+Current-buffer should be in `org-mode'."
+  (let ((denote-link-rx (rx "[[denote:")))
+    (while (re-search-forward denote-link-rx nil t)
+      (let* ((element (org-element-copy (org-element-context)))
+	     (linked-id (org-element-property :path element))
+	     (linked-file (denote-get-path-by-id linked-id))
+	     (linked-title (denote-retrieve-title-value linked-file 'org))
+	     (logseq-link (format "[[%s]]" linked-title)))
+	(replace-region-contents
+	 (org-element-property :begin element)
+	 (org-element-property :end element)
+	 (lambda () (format "%s " logseq-link)))
+	(org-fill-paragraph)))))
+
+(defun nm--migrate-denote-file-to-logseq (file)
+  "Convert denote note FILE to logseq.
+Copy the file from `denote-directory' to `logseq-directory' and
+convert all denote links to logseq."
+  (let ((dest (expand-file-name
+	       (file-name-nondirectory file)
+               logseq-directory)))
+    (with-temp-buffer
+      (insert-file-contents file)
+      (org-mode)
+      (nm--convert-denote-links-to-logseq)
+      (write-file dest))))
+
+(defun nm--ensure-dir-path (path)
+  "Ensure that PATH is a directory path.
+A path is directory path if it ends with a \"/\". Might not work
+on windows."
+  (if (string-match-p (rx "/" line-end) path)
+      path
+    (concat path "/")))
+
+(defun nm--migrate-denote-to-logseq ()
+  "Migrate all denote notes logseq.
+Migrate all org files in `denote-directory' to logseq notes in
+`logseq-directory'."
+  (let* ((denote-directory (nm--ensure-dir-path denote-directory))
+         (logseq-directory (nm--ensure-dir-path logseq-directory))
+         (denote-org-files (directory-files denote-directory t "\.org$")))
```

Comment:
Do logseq support non-org files as well? Do people have things like images and other files in the `denote-directory`? If yes, I think we should handle those as well by simply copying them over to `logseq-directory`.

Reply:
>  I think I would prefer to postpone this feature and wait for people to request and/or implement it.

Seems reasonable.